index.d.ts

```diff
@@ -0,0 +1,98 @@
+// Type definitions for @node-saml/xml-crypto
+// Project: https://github.com/node-saml/xml-crypto#readme
+// Original definitions by: Eric Heikes <https://github.com/eheikes>
+//                          Max Chehab <https://github.com/maxchehab>
+
+/// <reference types="node" />
+
+import { SelectedValue } from 'xpath';
+
+export class HashAlgorithm {
+    getAlgorithmName(): string;
+    getHash(xml: string): string;
+}
+
+export interface Reference {
+    xpath: string;
+    transforms?: ReadonlyArray<string> | undefined;
+    digestAlgorithm?: string | undefined;
+    uri?: string | undefined;
+    digestValue?: string | undefined;
+    inclusiveNamespacesPrefixList?: string | undefined;
+    isEmptyUri?: boolean | undefined;
+}
+
+export class SignatureAlgorithm {
+    getAlgorithmName(): string;
+    getSignature(signedInfo: Node, signingKey: Buffer): string;
+}
+
+export class TransformationAlgorithm {
+    getAlgorithmName(): string;
+    process(node: Node): string;
+}
+
+export class SignedXml {
+    static CanonicalizationAlgorithms: {[uri: string]: new () => TransformationAlgorithm };
+    static HashAlgorithms: {[uri: string]: new () => HashAlgorithm};
+    static SignatureAlgorithms: {[uri: string]: new () => SignatureAlgorithm};
+    canonicalizationAlgorithm: string;
+    keyInfoProvider: KeyInfo;
+    references: Reference[];
+    signatureAlgorithm: string;
+    signingKey: Buffer | string;
+    validationErrors: string[];
+    constructor(idMode?: string | null, options?: {
+        canonicalizationAlgorithm?: string | undefined
+        idAttribute?: string | undefined
+        implicitTransforms?: ReadonlyArray<string> | undefined
+        signatureAlgorithm?: string | undefined
+    })
+    addReference(
+        xpath: string,
+        transforms?: ReadonlyArray<string>,
+        digestAlgorithm?: string,
+        uri?: string,
+        digestValue?: string,
+        inclusiveNamespacesPrefixList?: string,
+        isEmptyUri?: boolean
+    ): void;
+    checkSignature(xml: string): boolean;
+    computeSignature(
+        xml: string,
+        opts?: {
+            prefix?: string | undefined,
+            attrs?: {[key: string]: any} | undefined,
+            location?: {
+                reference: string,
+                action: 'append' | 'prepend' | 'before' |  'after'
+            } | undefined,
+            existingPrefixes?: {[prefix: string]: string} | undefined
+        }
+    ): void;
+    getOriginalXmlWithIds(): string;
+    getSignatureXml(): string;
+    getSignedXml(): string;
+    loadSignature(signatureNode: string | Node): void;
+}
+
+export interface KeyInfo {
+    getKey(keyInfo?: Node): Buffer;
```

Comment:
Ha! I brought this issue up to the DT maintainer: getKey should be accepting an array of Nodes, not a single Node object

I will link the issue and explanation here in a moment

Reply:
So something like...
```suggestion
    getKey(keyInfo?: null | Node[]): Buffer;
```
Per that discussion?